Epoch 0:   0%|                                                                                  | 0/15 [00:00<?, ?it/s]
/home/fathinah.izzati/TJ/filtered/tnj_1/rvq/CopyofCopyof051-SafetySecond_clip_037_360-370.npy
motion torch.Size([500, 2, 499200])
torch.Size([1, 251, 2, 499200]) torch.Size([1, 251, 1]) torch.Size([1, 251, 1]) torch.Size([1, 2, 251])
o.shape torch.Size([1, 251, 2, 2048])
cond shape torch.Size([1, 251, 2, 48])
merge linear torch.Size([1, 251, 2, 2048])
self.pos_emb[i + 1] torch.Size([251, 2, 2048])
self.pos_emb[i + 1][None, :T] torch.Size([1, 251, 2, 2048])
self.pos_emb[i + 1][None, :T].repeat(B, 1, 1,1) torch.Size([1, 251, 2, 2048])
embedding shape torch.Size([1, 251, 2, 2048])
Traceback (most recent call last):
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/einops/einops.py", line 522, in reduce
    recipe = _prepare_transformation_recipe(pattern, reduction, axes_names=tuple(axes_lengths), ndim=len(shape))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/einops/einops.py", line 365, in _prepare_transformation_recipe
    raise EinopsError(f"Wrong shape: expected {len(left.composition)} dims. Received {ndim}-dim tensor.")
einops.EinopsError: Wrong shape: expected 3 dims. Received 4-dim tensor.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/l/users/fathinah.izzati/expotion_new/train.py", line 235, in <module>
    main(args)
  File "/l/users/fathinah.izzati/expotion_new/train.py", line 210, in main
    train(
  File "/l/users/fathinah.izzati/expotion_new/train.py", line 149, in train
    outputs = model(**batch_1)
              ^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/l/users/fathinah.izzati/expotion_new/coco_mulla/models/model_motion.py", line 310, in forward
    embed_fn = self.cp_transformer(face=face,
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/l/users/fathinah.izzati/expotion_new/coco_mulla/models/model_motion.py", line 257, in forward
    q, k, v, o = self.layers[i](o, embedding)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/l/users/fathinah.izzati/expotion_new/coco_mulla/models/model_motion.py", line 158, in forward
    q, k, v, o = self.self_attn(nx, nx, nx, emb_fn=None,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/l/users/fathinah.izzati/expotion_new/coco_mulla/models/transformer_cc.py", line 388, in forward
    packed = rearrange(projected, f"b t (p h d) -> {bound_layout}", p=3, h=self.num_heads)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/einops/einops.py", line 591, in rearrange
    return reduce(tensor, pattern, reduction="rearrange", **axes_lengths)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fathinah.izzati/miniconda3/envs/cocomulla/lib/python3.11/site-packages/einops/einops.py", line 533, in reduce
    raise EinopsError(message + "\n {}".format(e))
einops.EinopsError:  Error while processing rearrange-reduction pattern "b t (p h d) -> b h p t d".
 Input tensor shape: torch.Size([1, 251, 2, 6144]). Additional info: {'p': 3, 'h': 32}.
 Wrong shape: expected 3 dims. Received 4-dim tensor.
